companionHandler.js

```diff
@@ -22,9 +23,24 @@ function handleCompanion(problem) {
         const dir = vscode.workspace.workspaceFolders[0].uri.fsPath;
         const languageChoices = Object.keys(config.extensions);
 
-        vscode.window.showQuickPick(languageChoices, {
-            placeHolder: "Select the language"
-        })
+        const defaultLanguage = preferences().get("defaultLanguage");
+        let selectLanguage;
+        if(defaultLanguage != "") {
+            selectLanguage = new Promise( (resolve) => {
```

Comment:
Instead of using this useless promise, you can move both the logics within a single promise, and then do the clecking logic `defaultLanguage != "" ...` inside this single promise. This can also help to eliminate the variable `selectLanguage`.